Source/WebCore/crypto/gcrypt/CryptoKeyOKPGCrypt.cpp

```diff
@@ -151,17 +152,22 @@ std::optional<CryptoKeyPair> CryptoKeyOKP::platformGeneratePair(CryptoAlgorithmI
 
 bool CryptoKeyOKP::platformCheckPairedKeys(CryptoAlgorithmIdentifier, NamedCurve namedCurve, const Vector<uint8_t>& privateKey, const Vector<uint8_t>& publicKey)
 {
-    if (namedCurve != NamedCurve::Ed25519 && namedCurve != NamedCurve::X25519)
+    if (privateKey.size() != 32 || publicKey.size() != 32)
         return false;
 
-    if (namedCurve == NamedCurve::X25519) {
+    switch (namedCurve) {
+    case NamedCurve::X25519: {
         // public key being X25519(a, 9), as defined in [RFC7748], section 6.1.
         auto q = X25519(privateKey, WebCore::CryptoKeyOKPImpl::nine);
         return q && q->size() == 32 && *q == publicKey;
     }
+    case NamedCurve::Ed25519:
+        return validateEd25519KeyPair(privateKey, publicKey);
+    default:
+        ASSERT_NOT_REACHED();
+    }
 
-    // FIXME: Implement this check for Ed25519
-
+    // No check
     return true;
```

Comment:
By default, this should return false. That way the supported curves perform their own checks, and anything else would default to a failed check, regardless of the debug-specific assert.

Reply:
If we implement it this way, non-supported algorithms would pass the tests we have to ensure 2 incompatible keys are not imported.